File: app/src/main/java/com/github/hurshi/MainModule.java
package com.github.hurshi;

import com.github.hurshi.bean.Dog;
import com.github.hurshi.scopes.ActivityScope;

import dagger.Module;
import dagger.Provides;

/**
 * Created by gavin on 2017/10/24.
 */

@Module
public class MainModule {


    @ActivityScope
    @Provides
    public Dog provideDog() {
        return new Dog("旺财", 3);
    }


}

File: app_kotlin/src/main/java/com/github/hurshi/app_kotlin/MainModule.kt
package com.github.hurshi.app_kotlin

import com.github.hurshi.app_kotlin.bean.Dog
import com.github.hurshi.app_kotlin.scopes.ActivityScope

import dagger.Module
import dagger.Provides

/**
 * Created by gavin on 2017/10/24.
 */

@Module
class MainModule {


    @ActivityScope
    @Provides
    fun provideDog(): Dog {
        return Dog("旺财", 3)
    }


}
